android_10/src/org/ros/android/MasterChooser.java

```diff
@@ -111,16 +131,22 @@ public boolean hasStableIds() {
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.master_chooser);
+    final Pattern uriPattern = RosURIPattern.URI;
     uriText = (EditText) findViewById(R.id.master_chooser_uri);
     connectButton = (Button) findViewById(R.id.master_chooser_ok);
     uriText.addTextChangedListener(new TextWatcher() {
       @Override
       public void onTextChanged(CharSequence s, int start, int before, int count) {
-        if (s.length() > 0) {
-          connectButton.setEnabled(true);
-        } else {
-          connectButton.setEnabled(false);
-        }
+          final String uri = s.toString();
```

Comment:
very minor nit: I think there is an indentation issue here, right? Shouldn't it be two spaces? (to follow the rest of the repository's convention)